### ISSUE TITLE
Impossible to edit questions/answers with keyboard only after initial writing

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/forms#416]
1. Some a questions with some answers
2. Now don’t use your mouse anymore
3. Using the `Tab` key, try navigating using just your keyboard and edit the existing questions.

It’s not possible as you can not focus the titles or answers, and focusing them does not change it to editing mode.

@jotoeri @skjnldsv any idea how we can trigger that? We can for sure put `tabindex="0"` on each of the question and answer elements, but it needs to then trigger a switch to the edit view.

**Comment by elioqoshi** [comment]
This is an accessibility issue which blocks people with certain disabilities from using forms. I'd be glad to help if there is anything I can do.

**Comment by susnux** [comment]
Currently we use `@click` to enable the edit mode, I think we need an `@focus` additionally with `@focusout` to stop editing.

**Comment by fcnjd** [comment]
Hello,
thank you for tracking this issue. I as a user would like to provide an additional task item:
A part of editing questions is also to change the order. This as well isn't possible using only the keyboard, a mouse is required. It would be nice to have propperly labelled buttons for each question, which move it one position upwards or downwards accordingly.

**Comment by Chartman123** [comment]
@fcnjd Re-ordering the questions is also already tracked as a separate issue and we're currently working on it :) #317

**Comment by Chartman123** [comment]
@susnux @jancborchardt @jotoeri Provoking idea: What about always switching all fields to editable as long as we're in the edit view? This way we wouldn't have to deal with adding a focus trap to the non-input variants in the edit view.

**Comment by jotoeri** [comment]
Not so provoking, i'd say. I had that thought, too. Since we have the edit/submit now directly accessible, that would make several things much easier, if there is only the global edit mode.
So +1 from my side. :)

**Comment by susnux** [comment]
I am not that sure about this, as it is quite nice to have a preview of the form while creating it.
Its at least one of the nice UI features we have, don't you think so?

**Comment by Chartman123** [comment]
Except for the multiple choice questions it's mostly no difference currently. Only the markdown fields will have a bigger difference between edit/view.

But on the other side I remember all the hassle about aligning the layout in edit/view modes everytime there are bigger design changes in the server core...

**Comment by Chartman123** [comment]
> nice to have a preview of the form while creating it.

@susnux I can imagine to at least have some kind of tabbed input field for the descriptions to directly see the markdown rendering just like for example here on Github.
